### DISCUSSION TITLE
CloudKit compliant for CoreData models

### DISCUSSION THREAD

**Discussion by Hengyu** [carekit-apple/CareKit#652] [Ideas]
Hey team,

The current data models in CareKit is not CloudKit compliant, e.g., all fields should be nullable if the data model is configured for CloudKit.

My question is: do we have any plan to make the CareKit compliant with the CloudKit in the next major release?

Thanks!
Hengyu

**Comment by erik-apple** [comment]
CareKit is intentionally not compatible with CloudKit because it does not meet the strict legal requirements for storing medical data in certain countries.

If you intend to move CareKit data off device, be sure to carefully research your cloud solution to ensure that it meets all the legal requirements that apply in each of the countries you’re planning to ship to.

**Reply by Hengyu** [reply]
Thanks Eric for the explanations!
But still wondering if users can sync their HealthKit data with their iCloud account why CloudKit is not legal compatible.

**Comment by nizar** [comment]
@erik-apple @Hengyu I am currently working on adapting CareKitStore to Firestore, which appears to be HIPAA compliant (https://cloud.google.com/security/compliance/hipaa/). If this is something others are interested in, I'd love to collaborate.

**Comment by marco-theraforge** [comment]
We have implemented AWS support for CK and ResearchKit, with IBM Cloud, Azure and other cloud SPs (including on premise) in the works:
https://github.com/TheraForge/Getting-Started
If anyone is interested, you can follow/contact us on Github or here: [www.hippocratestech.com](https://www.hippocratestech.com)
